### PR TITLE
Feature/simulation

### PR DESCRIPTION
**Pull request by hiagolcm**
#### What problem is this solving?
The search-engines have a performance problem since they need to call the simulation API for each product. This PR allows setting a prop to call this API on the client-side.

I also changed the SVG spinner to a CSS spinner. The SVG one was causing performance problems.

#### How to test it?

[Workspace](https://hiago3--carrefourbrfood.myvtex.com/Alimentos-e-Bebidas/Bebidas?crfint=hm-tlink|alimentos-e-bebidas|1|bebidas|2)

#### Screenshots or example usage:

Czffg3e94c

### CODE REVIEW

**Review by iaronaraujo — APPROVED** [review]

File: react/hooks/useIsPriceAsync.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useSearchPage } from 'vtex.search-page-context/SearchPageContext'
+
+const useIsPriceAsync = () => {
+  const { searchQuery } = useSearchPage()
```
Comment (klzns) [inline react/hooks/useIsPriceAsync.js:4]: Can this destructure break if it's outside of search? I mean, if useSearchPage() returns undefined, what happens?

File: react/hooks/useSimulation.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    variables: {
+      items: simulationItemsInput,
+    },
+    skip: !isPriceAsync || !inView,
+    ssr: false,
+    onError,
+    onCompleted: (response) => {
+      const simulationItems = response.itemsWithSimulation
+
+      const mergedProduct = clone(product)
```
Comment (klzns) [inline react/hooks/useSimulation.js:46]: We do we have to merge, can't we just update the whole product data?

**Review by klzns — CHANGES_REQUESTED** [review]

File: react/hooks/useIsPriceAsync.js (hunk 1 above)
Comment (hiagolcm) [reply react/hooks/useIsPriceAsync.js]: `useSearchPage` return `{}` when there is no context
anyway, I changed my code to be extra careful

File: react/hooks/useSimulation.js (hunk 2 above)
Comment (hiagolcm) [reply react/hooks/useSimulation.js:46]: Unfortunately, the simulation API doesn't return the whole item,  so I need to do this merge 😢

**Review by thalytafabrine — APPROVED** [review]

**Review by klzns — APPROVED** [review]

**Pull request merged**